diffrax/_solver/srk.py

```diff
@@ -0,0 +1,624 @@
+            elif stla:
+                if path.levy_area not in ["space-time", "space-time-time"]:
+                    raise ValueError(
+                        "The Brownian path controlling the diffusion "
+                        "should be initialised with `levy_area='space-time'`"
+                        "or `levy_area='space-time-time'`"
+                    )
+
+        if self.tableau.additive_noise:
+            # check that the vector field of the diffusion term does not depend on y
+            ones_like_y0 = jtu.tree_map(lambda x: jnp.ones_like(x), y0)
+            _, y_sigma = eqx.filter_jvp(
+                lambda y: diffusion.vf(t0, y, args), (y0,), (ones_like_y0,)
+            )
+            # check if the PyTree is just made of Nones (inside other containers)
+            if len(jtu.tree_leaves(y_sigma)) > 0:
+                raise ValueError(
+                    "Vector field of the diffusion term should be constant, "
+                    "independent of y."
+                )
+
+        return None
+
+    def _embed_a_lower(self, _a, dtype):
+        num_stages = len(self.tableau.b_sol)
+        tab_a = np.zeros((num_stages, num_stages))
+        for i, a_i in enumerate(_a):
+            tab_a[i + 1, : i + 1] = a_i
+        return jnp.asarray(tab_a, dtype=dtype)
+
+    def step(
+        self,
+        terms: term_structure,
+        t0: RealScalarLike,
+        t1: RealScalarLike,
+        y0: Y,
+        args: PyTree,
+        solver_state: _SolverState,
+        made_jump: BoolScalarLike,
+    ) -> tuple[Y, _ErrorEstimate, DenseInfo, _SolverState, RESULTS]:
+        del solver_state, made_jump
+
+        dtype = jnp.result_type(*jtu.tree_leaves(y0))
+        drift, diffusion = terms.terms
+        additive_noise = self.tableau.additive_noise
+        if self.tableau.ignore_stage_f is not None:
+            ignore_stage_f = jnp.array(self.tableau.ignore_stage_f)
+        if self.tableau.ignore_stage_g is not None:
+            ignore_stage_g = jnp.array(self.tableau.ignore_stage_g)
+
+        # time increment
+        h = t1 - t0
+
+        # First the drift related stuff
+        a = self._embed_a_lower(self.tableau.a, dtype)
+        c = jnp.asarray(np.insert(self.tableau.c, 0, 0.0), dtype=dtype)
+        b_sol = jnp.asarray(self.tableau.b_sol, dtype=dtype)
+
+        def make_zeros():
+            def make_zeros_aux(leaf):
+                return jnp.zeros((len(b_sol),) + leaf.shape, dtype=leaf.dtype)
+
+            return jtu.tree_map(make_zeros_aux, y0)
+
+        # _tfs is a PyTree of the same shape as y0, except that the arrays inside
+        # have an additional batch dimension of size len(b_sol) (i.e. num stages)
+        # This will be one of the entries of the carry of lax.scan. In each stage
+        # one of the zeros will get replaced by the value of
+        # h * f(t0 + c_j * h, z_j) where z_j is the jth stage of the SRK.
+        # The name tf is because the value of f(t0 + c_j * h, z_j) is already
```

Comment:
I think your `f` on this line is missing a `h *`.

By the way, can we call this array `ks`? This is just the convention used in the existing RK solvers for ODEs for representing `f * h`.

Reply:
I'd prefer to keep it as `tf`, because that lets us differentiate between the f stages and the g stages. And also to make it clear it is already multiplied by the time increment. And I try to avoid an h in variable names because that can be confused with stla.

Reply:
I think lets go for `ks_f` and `ks_g` (or `ks_drift` and `ks_diffusion` might be even better) for consistency with `AbstractRungeKutta`, then.

Reply:
Sounds good, will do.